### PR TITLE
Added note for Service Loader.

### PR DESCRIPTION
**Pull request by JulianFeinauer**
I'm sorry for this small PR but I wanted to document the result of our discussion (also for myself) and I cant push directly to master : )

### CODE REVIEW

**Comment by JulianFeinauer** [comment]
@sruehl thanks for the hint, good catch.
I have problems mit mvn site:run and filed jira PLC4X-52.

File: plc4j/api/src/main/java/org/apache/plc4x/java/api/PlcDriver.java [hunk 1]
```diff
@@ -24,6 +24,10 @@ Licensed to the Apache Software Foundation (ASF) under one
 
 /**
  * General interface defining the minimal methods required for adding a new type of driver to the PLC4J system.
+ *
+ * <b>Note that each driver has to add a service file called org.apache.plc4x.java.PlcDriver to
+ * src/main/resources/META-INF which contains the fully qualified classname in order to get loaded
+ * by the {@link java.sql.DriverManager} instances.</b
```
Comment (sruehl) [inline plc4j/api/src/main/java/org/apache/plc4x/java/api/PlcDriver.java]: `</b` is missing the closing `>`

Comment (sruehl) [reply plc4j/api/src/main/java/org/apache/plc4x/java/api/PlcDriver.java]: that might be where the first has gone it the first place :)

**Comment by JulianFeinauer** [comment]
Oh my... that was a classical "i fix this in a second and then...".
Sorry for that and thanks for the catch. I will ask chris on the list to look into the maven problems (if he doesnt already) to be able to run "tests"

File: plc4j/api/src/main/java/org/apache/plc4x/java/api/PlcDriver.java [hunk 2]
```diff
@@ -24,6 +24,10 @@ Licensed to the Apache Software Foundation (ASF) under one
 
 /**
  * General interface defining the minimal methods required for adding a new type of driver to the PLC4J system.
+ *
+ * <b>Note that each driver has to add a service file called org.apache.plc4x.java.PlcDriver to
+ * src/main/resources/META-INF which contains the fully qualified classname in order to get loaded
+ * by the {@link java.sql.DriverManager} instances.</b>
```
Comment (sruehl) [inline plc4j/api/src/main/java/org/apache/plc4x/java/api/PlcDriver.java]: `java.sql.DriverManager` shouldn't that be the `org.apache.plc4x.java.PlcDriverManager`?

**Comment by JulianFeinauer** [comment]
@sruehl I noticed that I wasnt able to reference the PlcDriverManager as it is placed in core. So I suggest to extract an interface PlcDriverManager and to rename the implementation to DefaultPlcDriverManager.
The Interface is especially necessary for us to implement our proxy, as the PlcDriverManager can be used as an equivalent of the javax.sql Datasource.

**Comment by sruehl** [comment]
> @sruehl I noticed that I wasnt able to reference the PlcDriverManager as it is placed in core. So I suggest to extract an interface PlcDriverManager and to rename the implementation to DefaultPlcDriverManager.
> The Interface is especially necessary for us to implement our proxy, as the PlcDriverManager can be used as an equivalent of the javax.sql Datasource.

fair enough im just thinking if there would be a better way to retrieve the PlcDriverManager as to use the `DefaultPlcDriverManager`. Might confuse a Api User but im not sure what would be best though...

**Comment by JulianFeinauer** [comment]
I agree, and I also did not like that.
I shortly thought about addind a static factory method to the interface but this is something that I dont like that much.
I'm not sure what the best way is as it feels right to have the interface in the api module but I agree that it is confusing for users.

**Comment by sruehl** [comment]
What we could do I add a Interface on top of the existing driver manager like `IPlcDriverManager` or something and write in the jdoc that it needed for proxies. This way the usage stays the same and the purpose is clear. But dunno :/

**Comment by JulianFeinauer** [comment]
I'm also unsure about that.
We can also leave it as is but perhaps we should move the PlcDriverManager to the api package?

Another approach would be to use a PlcDriverManagerFactory witha  static create method (kind of like the entity manager).

**Comment by chrisdutz** [comment]
I would have left the DriverManager (But I do like the name PlcDriverManager more) in the API module, but the problem was that at that time I couldn't write any tests for that as that would have required a cyclic dependency in the maven project graph. I definitely agree that it would be cooler to just depend on the API module. Even if the DriverManager isn't really an API class ... but I guess users would probably forgive us that.

**Comment by sruehl** [comment]
the factory `PlcDriverManagerFactory` approach might be the best at the end... but keep in mind that this factory can't the reference the default one directly as we would get an circular dependency

**Comment by JulianFeinauer** [comment]
Yes... this is kind of tricky.
A kingdom for some injection frameworks.. : )
I suggest to revert my changes and remove the link in the javadoc to make this merge-able.
Then we discuss the topic seperate, what do you think?

**Comment by sruehl** [comment]
good, idea. Maybe you could squash/soft-reset your changes into one commit.

**Comment by JulianFeinauer** [comment]
@chrisdutz So you suggest to move the PlcDriverManager to the api module to resolve the problem?

**Comment by sruehl** [comment]
> @chrisdutz So you suggest to move the PlcDriverManager to the api module to resolve the problem?

as far as I Understand we wouldn't have the cyclic dependency problem then and you could reference the default one from a factory and still use the interface for your proxy...

**Comment by chrisdutz** [comment]
I think things changed with the latest refactoring of the API as we no longer have real classes, just interfaces ... I guess we could try moving that class there and get rid of the "core" module. Would definitely be worth it. I just knew that I wanted it in the API but I had to introduce the core in order to resolve the cyclic dependencies (At that time ... might be obsolete now)

**Pull request merged**